File: Tree/461-getAllElementsForTwoBST.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution(object):
    def getAllElements(self, root1, root2):
        """
        :type root1: TreeNode
        :type root2: TreeNode
        :rtype: List[int]
        """
        def inorder(root):
            if not root:
                return 
            return inorder(root.left)+[root.val]+inorder(root.right)
        
        a=inorder(root1)
        b=inorder(root2)
        la,lb=len(a),len(b)
        i,j=0,0
        ans=[]
        while i<la and j<lb:
            if a[i]<=b[j]:
                ans.append(a[i])
                i+=1
            else:
                ans.append(b[j])
                j+=1
        if i<la:
            ans.extend(a[i:])
        if j<lb:
            ans.extend(b[j:])
        return ans
        

File: Tree/461-getAllElementsForTwoBST.cpp
#include<iostream>
#include<vector>

using VEC=std::vector<int>;


class TreeNode
{
    public:
        int val{};
        TreeNode* left{};
        TreeNode* right{};
        TreeNode(int v,TreeNode* l=nullptr,TreeNode* r=nullptr):val{v},left{l},right{r}{};
};

void inorder(TreeNode* r,VEC& list)
{
    if(!r)
        return;
    inorder(r->left,list);
    list.push_back(r->val);
    inorder(r->right,list);
}

VEC allElements(TreeNode* r1,TreeNode* r2)
{
    VEC ans{};
    VEC l1{},l2{};
    
    inorder(r1,l1);
    inorder(r2,l2);

    //merge values
    int n1=l1.size(),n2=l2.size();
    int i=0,j=0;
    while(i<n1 &&j<n2)
    {
        if(l1[i]<l2[j])
        {
            ans.push_back(l1[i]);
            i+=1;
        }
        else
        {
            ans.push_back(l2[j]);
            j+=1;
        }
    }
    if(i<n1)
    {
        for(auto &v:l1)
            ans.push_back(v);
    }
    if(j<n2)
    {
        for(auto &v:l2)
            ans.push_back(v);
    }
    return ans;
}

int main()
{
    TreeNode* r1=new TreeNode(2);
    TreeNode* l11=new TreeNode(1);
    TreeNode* r11=new TreeNode(3);

    r1->right=r11;
    r1->left=l11;

    TreeNode* r2=new TreeNode(5);
    TreeNode* l22=new TreeNode(4);
    TreeNode* r22=new TreeNode(6);
    r2->left=l22;
    r2->right=r22;

    VEC res=allElements(r1,r2);
    for(auto &v:res)
        std::cout<<v<<',';
}